hosting/hosting_test.go

```
@@ -100,6 +101,15 @@ func TestListLocalAssetMetadata(t *testing.T) {
 
 	u.So(t, assetMetadata, gc.ShouldResemble, testData)
 
+	assetDescriptions = map[string]hosting.AssetDescription{
+		path3: {
+			FilePath: path3,
+			Attrs:    []hosting.AssetAttribute{jsonAttr},
+		},
+	}
+	assetMetadata, listErr = hosting.ListLocalAssetMetadata(appID, rootDir, assetDescriptions, assetCache)
+	u.So(t, listErr, gc.ShouldNotBeNil)
```

nit: we should make this assertion more precise so that other unrelated failures don't cause this test to silently pass if the behavior changes but it still returns a (different) error.
You could add a check that `listErr.Error()` matches the string of the expected error message.